internal/ethapi/api.go

```diff
@@ -1092,7 +1092,7 @@ func (s *BlockChainAPI) GetBlockReceipts(ctx context.Context, blockNrOrHash rpc.
 
 	result := make([]map[string]interface{}, len(receipts))
 	for i, receipt := range receipts {
-		result[i] = marshalReceipt(receipt, block.Hash(), block.NumberU64(), signer, txs[i], i)
+		result[i] = marshalReceipt(receipt, block.Hash(), block.NumberU64(), signer, txs[i], i, false)
 	}
```

Comment:
Hi! Should we append state sync receipt after the expected receipts?

```suggestion
	result := make([]map[string]interface{}, len(receipts+1))
	for i, receipt := range receipts {
		result[i] = marshalReceipt(receipt, block.Hash(), block.NumberU64(), signer, txs[i], i, false)
        }
        
        receipt := rawdb.ReadBorReceipt(api.eth.chainDb, blockHash, blockNumber.Uint64(), api.eth.blockchain.Config())
        if receipt != nil {
            tx, _, _, _ := rawdb.ReadBorTransaction(api.eth.chainDb, receipt.TxHash)
            fields := ethapi.MarshalReceipt(receipt, blockHash, blockNumber.Uint64(), signer, tx, int(receipt.TransactionIndex), true)

            txReceipts = append(txReceipts, fields)
       }
```